src/platform/pic32/uart.cpp

```diff
@@ -163,7 +181,32 @@ bool openxc::interface::uart::connected(UartDevice* device) {
             status = false;
             break;
     }
-    return status;
 
 #endif
+
+	// on apple devices, it takes some time for flow control to propagate to the device
```

Comment:
Very interesting - is this the source of the bug where it would disconnect after ~5 seconds?